### PR TITLE
feat(microservices): log event name w/o handler

### LINKED ISSUE #5812: Log event name when no handler is registered for it
## Feature Request

Include event name in the error message when no handler is registered for that event.

## Is your feature request related to a problem? Please describe.

If a microservice receives an event which doesn't have any handler, it'd be super helpful to know what the event name(pattern) was in the logs.

For example, I receive the following error log when a `foo` event is published to a queue and my app has no handler for it. You can see it's pretty hard to figure what I'm missing in my app configurations.

Queue message content:
```json
{ "pattern": "foo", "data": { }  }
```

Error log:
```text
[Nest] 1428   - 11/27/2020, 4:07:07 PM   [Server] There is no matching event handler defined in the remote service. +7468ms
```

## Describe the solution you'd like

Events, or even other types such as Messages, that do not have any handlers cause the server to write an error log. We can expand that by providing the pattern name in the log.

https://github.com/nestjs/nest/blob/543432cce9453cdd12677a06e1dd3ba831c8a15d/packages/microservices/server/server.ts#L96-L99

So the error log could have a bit more info to help me debug this issue:

```text
[Nest] 1428   - 11/27/2020, 4:07:07 PM   [Server] There is no matching event handler defined for event pattern "foo" in the remote service. +7468ms
```

## What is the motivation / use case for changing the behavior?

Easier discoverability of configuration issues. Especially for newcomers to the Nest.

### PR DESCRIPTION
**Pull request by poulad**
A descriptive error message tells user exactly what event pattern is missing a handler.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: Fixes #5812

## What is the new behavior?
A more descriptive error message. See the discussion on issue.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: packages/microservices/server/server.ts [hunk 1]
```diff
@@ -95,7 +95,9 @@ export abstract class Server {
   ): Promise<any> {
     const handler = this.getHandlerByPattern(pattern);
     if (!handler) {
-      return this.logger.error(NO_EVENT_HANDLER);
+      return this.logger.error(
+        `${NO_EVENT_HANDLER} Event pattern is ${JSON.stringify(pattern)}.`,
```
Comment (jmcdo29) [inline packages/microservices/server/server.ts:99]:
```suggestion
        `${NO_EVENT_HANDLER} Event pattern: ${JSON.stringify(pattern)}.`,
```

**Review by jmcdo29 — CHANGES_REQUESTED** [review]

File: packages/microservices/server/server.ts (hunk 1 above)
Comment (jmcdo29) [reply packages/microservices/server/server.ts:99]: Either this or `Event pattern requested: ...`

**Review by jmcdo29 — APPROVED** [review]

**Comment by poulad** [comment]
hey @kamilmysliwiec 
Let me know if any concerns with this PR.

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**